app/views/idv/otp_verification/show.html.erb

```diff
@@ -36,11 +36,14 @@
   </div>
 <% end %>
 
-<%= button_to idv_resend_otp_path,
-              method: :post,
-              class: 'usa-button usa-button--outline ico ico-refresh margin-bottom-4' do %>
-  <%= t('links.two_factor_authentication.get_another_code') %>
-<% end %>
+<%= render ButtonComponent.new(
+      idv_resend_otp_path,
+      method: :post,
+      factory: :button_to,
+      outline: true,
+      icon: :loop,
+      class: 'margin-bottom-4',
+    ).with_content(t('links.two_factor_authentication.get_another_code')) %>
```

Comment:
Noodling on the `factory` bit here, I feel like that name doesn't quite cover what's going on, and `factory_args` only makes sense when looking at examples.

Here's an idea that uses procs?


```suggestion
<%= render ButtonComponent.new(
      idv_resend_otp_path,
      action: proc do |body, url, options|
        button_to(body, url, method: :post, **options)
      end,
      factory: :button_to,
      outline: true,
      icon: :loop,
      class: 'margin-bottom-4',
    ).with_content(t('links.two_factor_authentication.get_another_code')) %>
```

 We could have a default value for the proc that serves as an example:

```ruby
class ButtonComponent
  def initialize(url, action: proc { |body, url, options| link_to(body, url, options) },  **options)
  # ... etc
   end
end
```

Reply:
or honestly, we just call it `link_or_button: :link` because realistically there are only those two factories we are choosing from?

Then all the stuff into `link_or_button_options` and `public_send(link_or_button, content, url_link_or_button_options)`

Reply:
> or honestly, we just call it `link_or_button: :link` because realistically there are only those two factories we are choosing from?
> 
> Then all the stuff into `link_or_button_options` and `public_send(link_or_button, content, url_link_or_button_options)`

We've got `button_to` and `button_tag` within these changes already, and I expect we'd probably also want to support SimpleForm `f.button` and `f.submit` as well... though it's doubtful those could be supported with the implementation as it stands.

I like the proc idea. It's a bit more verbose, but with a reasonable default I think it can also be clearer to understand how to customize, and also more flexible to support the SimpleForm cases as well.

Reply:
Took a pass at it in 2c1e7aaad. WDYT?

Reply:
I like it! I didn't even think about SimpleForm, let's ship the proc 🚢